libmscore/barline.cpp

```diff
@@ -616,7 +616,8 @@ void BarLine::drawEditMode(QPainter* p, EditData& ed)
       BarLineEditData* bed = static_cast<BarLineEditData*>(ed.getData(this));
       y1 += bed->yoff1;
       y2 += bed->yoff2;
-      QPointF pos(pagePos());
+      QPointF pos(canvasPos());
+      pos.ry() += measure()->system()->staff(staffIdx())->y();
```

Comment:
Wouldn't it be better to overload `canvasPos` for `BarLine` correctly instead?

Reply:
Maybe so.  I'm not entirely sure why the normal canvasPos() isn't correct and assumed it was something unique to when this is called during edit mode,  I think we used to rely on canvasPos more than we do know, and I figured lots of things would have been broken if canvasPos was.  But investigating further, it seems maybe were not actually using camvasPos for barlines anywhere else.  I'll do some testing and update if appropriate.